ruby_review_week_four.rb

```diff
@@ -2,36 +2,119 @@
 
+an instance variable is an attribute of an object
+
 # 6. Choose an everyday item (cup, computer, person) and give it two instance variables (which will be attributes), and one method (to describe a function of the object). Code the initialize method in full (for these attributes only), and the definition line of the instance method (but do not code the entire method).
 
+class Computer
+  def initialize(brand, year)
+  @brand = brand
+  @year = year
+  end
+
+  mollyscomp = Computer.new(brand: 'Macbook', year: 2011)
+
+  def run_macos?
+  end
+
+
 # 7. What is handy about using gems in Ruby?
 
+they provide additional functionality that is not inbuilt to Ruby ie adding colors, timers, geocoding, etc
+
 # 8. Define a class called fish, and leave it empty.
 
+class Fish
+  def initialize
+  end
+
 # 9. How would you use that class to make three fish objects?
 
+wanda = Fish.new
+rainbow = Fish.new
+koi = Fish.new
+
 # 10. Add two attributes to your fish class, name and species, and make them accessible for reading and writing.
 
+def initalize(name, species)
+  attr_accessor
+  @name = name
+  @species = species
+end
+
+
 # 11. 
 
 # a. Create one fish object and store it in a variable, and 
 
+fishy=Fish.new
+
+
 # b. on a second line, give that fish object’s name attribute the value of ‘Fred’ (assume you have access).
 
+fishy=Fish.new(name:"fred")
+
 # 12. Define a hash called ‘student’. This hash should have two keys, one called name, and the other called age, and assign a string to the name key, and an integer to the age key. 
 
+student = {name:"Molly", age:30}
+
+
 # 13. Write a Ruby method that takes three arguments and returns the sum (total) of those three arguments.
 
+def add_method(x,y,z)
+  sum = x+y+z
+  return sum
+end
+
 # 14. Write a Ruby method that takes an array and a string as arguments, and returns true if the string is in the array, and false if it is not.
 
-# 15. Write a method that takes one argument, a number, and that returns true if the number is divisible by three, and false otherwise. 
+def include(array,string)
+  if array.include?(string) == true
+    return true
+  else
+    return false
+  end
+end
+
+
+15. Write a method that takes one argument, a number, and that returns true if the number is divisible by three, and false otherwise. 
+
+def divbythree(x)
+  if x % 3 == 0
+    # puts "true"
+    return true
+  else
+    # puts "false"
+    return false
+  end
+end
+
+
+# 16. You must use your previous method in this next question. Write another method that takes an array (of numbers) as an argument. This method will use your first method to return two arrays (within one array - an array of arrays). The first array contains the numbers that are divisible by 3, and the other contains the rest of the numbers. For example, if you pass the second method the array [9, 4, 3, 5, 30, 2, 8] it would return [[9, 3, 30],[4, 5, 2, 8]].
+
+
+
+def arrayception(array)
+  array.partition do |i| 
```

Comment:
i haven't seen the use of the `partition` method before but this looks like it would work!